src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -1748,102 +1817,95 @@ private void PassivateCompleted(EntityId entityId)
             }
         }
 
-        private void DeliverMessage(object msg, IActorRef snd)
+        private void DeliverMessage(string entityId, object msg, IActorRef snd)
         {
-            var t = _extractEntityId(msg);
-            var entityId = t.Value.Item1;
-            var payload = t.Value.Item2;
+            var payload = _extractor.EntityMessage(msg)!; // payload can't be null unless dev really screwed up
 
-            if (string.IsNullOrEmpty(entityId))
-            {
-                Log.Warning("{0}: Id must not be empty, dropping message [{1}]", _typeName, msg.GetType().Name);
-                Context.System.DeadLetters.Tell(new Dropped(msg, "No recipient entity id", snd, Self));
-            }
-            else
+            if (payload is ShardRegion.StartEntity start)
```

Comment:
Ah, this is why we don't explicitly try to handle `ShardRegion.StartEntity` inside the wrapped message extractor - it gets handled here inside the `Shard` itself.